use-cases/inferencing/serving-with-vllm/README.md

```diff
@@ -0,0 +1,504 @@
+- Multi-Node Multi-GPU
+
+In this guide, you will serve a fine-tuned Gemma large language model (LLM) using graphical processing units (GPUs) on Google Kubernetes Engine (GKE) with the vLLM serving framework with the above mentioned deployment strategies. You can choose to swap the Gemma model with any other fine-tuned or instruction based model for inference on GKE.
+
+- Single GPU (no distributed inference) - If your model fits in a single GPU, you probably don’t need to use distributed inference. Just use the single GPU to run the inference.
+
+- Single-Node Multi-GPU (tensor parallel inference) - If your model is too large to fit in a single GPU, but it can fit in a single node with multiple GPUs, you can use tensor parallelism. The tensor parallel size is the number of GPUs you want to use. For example, if you need 4 GPUs, you can set the tensor parallel size to 4.
+
+By the end of this guide, you should be able to perform the following steps:
+
+- Create a Persistent Disk for the LLM model weights
+- Deploy a vLLM container to your cluster to host your model
+- Use vLLM to serve the fine-tuned Gemma model
+- View Production metrics for your model serving
+- Use custom metrics and Horizontal Pod Autoscaler (HPA) to scale your model
+
+## Prerequisites
+
+- This guide was developed to be run on the [playground AI/ML platform](/platforms/gke-aiml/playground/README.md). If you are using a different environment the scripts and manifest will need to be modified for that environment.
+- A bucket containing the fine-tuned model from the [Fine-tuning example](/use-cases/model-fine-tuning-pipeline/fine-tuning/pytorch/README.md)
+
+## Preparation
+
+- Clone the repository and change directory to the guide directory
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/accelerated-platforms && \
+  cd accelerated-platforms/use-cases/inferencing/serving-with-vllm
+  ```
+
+- Ensure that your `MLP_ENVIRONMENT_FILE` is configured
+
+  ```sh
+  cat ${MLP_ENVIRONMENT_FILE} && \
+  source ${MLP_ENVIRONMENT_FILE}
+  ```
+
+  > You should see the various variables populated with the information specific to your environment.
+
+- Set environment variables
+
+  ```sh
+  PROJECT_ID=your-project-id>
+  PROJECT_NUMBER=$(gcloud projects describe ${PROJECT_ID} --format="value(projectNumber)")
+  V_MODEL_BUCKET=<model-artifacts-bucket>
+  CLUSTER_NAME=<your-gke-cluster>
+  NAMESPACE=ml-team
+  KSA=<k8s-service-account>
+  HF_TOKEN=<your-Hugging-Face-account-token>
+  MODEL_ID=<your-model-id>
+  REGION=<your-region>
+  IMAGE_NAME=<your-image-name>
+  DISK_NAME=<your-disk-name>
+  ZONE=<your-disk-zone>
+  ACCELERATOR_TYPE=<accelerator_type> # nvidia-l4 | nvidia-tesla-a100
+  ```
+
+- Get Credentials for the GKE cluster
+
+  ```sh
+  gcloud container fleet memberships get-credentials ${MLP_CLUSTER_NAME} --project ${MLP_PROJECT_ID}
+  ```
+
+- Grant permission to kubernetes service account in cluster to access the storage bucket to view model weights
```

Comment:
platform playground can create the bucket

Reply:
This one is to create permissions for the ksa to access the storage bucket so it can download model weights. This is not creating the bucket only iam bindings and permission level

Reply:
platform sets permissions as well

Reply:
Yeah, an end user wouldn't have permissions to set any IAM. We should document any IAM requirements and we can leave it in for now until it is migrated to the playground.